packages/server/api/src/app/ee/platform-billing/platform-billing.service.ts

```diff
@@ -0,0 +1,68 @@
+import { ApSubscriptionStatus, DEFAULT_FREE_PLAN_LIMIT, PlatformBilling } from '@activepieces/ee-shared'
+import { apId, isNil } from '@activepieces/shared'
+import { FastifyBaseLogger } from 'fastify'
+import { repoFactory } from '../../core/db/repo-factory'
+import { platformService } from '../../platform/platform.service'
+import { userService } from '../../user/user-service'
+import { PlatformBillingEntity } from './platform-billing.entity'
+import { stripeHelper } from './stripe-helper'
+import Stripe from 'stripe'
+
+const platformBillingRepo = repoFactory(PlatformBillingEntity)
+
+export const platformBillingService = (log: FastifyBaseLogger) => ({
+    async getOrCreateForPlatform(platformId: string): Promise<PlatformBilling> {
+        const platformBilling = await platformBillingRepo().findOneBy({ platformId })
+        if (isNil(platformBilling)) {
+            return createInitialBilling(platformId, log)
+        }
+        return platformBilling
+    },
+
+    async update(platformId: string, tasksLimit: number | undefined, aiCreditsLimit: number | undefined): Promise<PlatformBilling> {
+        const platformBilling = await platformBillingRepo().findOneBy({ platformId })
+        if (isNil(platformBilling)) {
+            throw new Error('Platform billing not found')
+        }
+
+        if (!isNil(tasksLimit)) {
+            platformBilling.tasksLimit = tasksLimit
+        }
+
+        if (!isNil(aiCreditsLimit)) {
+            platformBilling.aiCreditsLimit = aiCreditsLimit
+        }
+
+        return platformBillingRepo().save(platformBilling)
+    },
```

Comment:
you can use spreadIfDefined from @activepieces/shared

```suggestion
    async update(platformId: string, tasksLimit: number | undefined, aiCreditsLimit: number | undefined): Promise<PlatformBilling> {
        const platformBilling = await platformBillingRepo().findOneBy({ platformId })
        if (isNil(platformBilling)) {
            throw new Error('Platform billing not found')
        }
        return platformBillingRepo().save({
            ...spreadIfDefined('tasksLimit', platformBilling.tasksLimit),
            ...spreadIfDefined('aiCreditsLimit', platformBilling.aiCreditsLimit),
        })
    },
```